controllers/metal3.io/baremetalhost_controller.go

```diff
@@ -216,6 +216,16 @@ func (r *BareMetalHostReconciler) Reconcile(ctx context.Context, request ctrl.Re
 	}
 	prov, err := r.ProvisionerFactory(*host, *bmcCreds, info.publishEvent)
 	if err != nil {
+
+		// Allow anyhow to proceed with the resource deletion if required
+		if !host.DeletionTimestamp.IsZero() && hostHasFinalizer(host) {
+			info.log.Info("marked to be deleted", "timestamp", host.DeletionTimestamp)
```

Comment:
Do we want to do this unconditionally?  What if the user wants the opportunity to fix e.g BMC credentials so that deprovisioning can be completed without errors?

Reply:
Isn't enough the check on DeletionTimeStamp? AFAIK, if set, it means that the client requested to delete the resource, but the process cannot be completed since the finalizer hasn't been removed yet.

Reply:
@andfasano you're correct
The danger here though is that we create a way for people to short-circuit the proper delete flow (i.e. skip deprovisioning). I commented on the previous PR for this:

> It would be safer if we checked the the Provisioning state was empty (i.e. we have never been able to create a provisioner for this Host, vs. we created a Host and provisioned it and now somebody has changed the URL).